scripts/build/builders/nuttx.py

```diff
@@ -0,0 +1,112 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from enum import Enum, auto
+
+from .gn import Builder
+
+
+class NuttXApp(Enum):
+    LIGHT = auto()
+
+    def ExampleName(self):
+        if self == NuttXApp.LIGHT:
+            return 'lighting-app'
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+    def AppNamePrefix(self, chip_name):
+        if self == NuttXApp.LIGHT:
+            return ('chip-%s-lighting-example' % chip_name)
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class NuttXBoard(Enum):
+    SIM = auto()
+
+
+def NuttXTarget(board, app):
+    if board == NuttXBoard.SIM:
+        if app == NuttXApp.LIGHT:
+            return 'sim:matter'
+    return 'none'
+
+
+class NuttXBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 app: NuttXApp = NuttXApp.LIGHT,
+                 board: NuttXBoard = NuttXBoard.SIM,
+                 enable_rpcs: bool = False,
+                 module_type: str = 'simulator',
+                 baudrate=2000000,
+                 enable_shell: bool = False,
+                 enable_cdc: bool = False,
+                 enable_resetCnt: bool = False,
+                 enable_rotating_device_id: bool = False,
+                 enable_ethernet: bool = False,
+                 enable_wifi: bool = False,
+                 enable_thread: bool = False,
+                 enable_frame_ptr: bool = False,
+                 enable_heap_monitoring: bool = False,
+                 use_matter_openthread: bool = False,
+                 ):
+
+        nuttx_chip = 'nuttx'
+
+        super(NuttXBuilder, self).__init__(
+            root=os.path.join(root, 'examples',
+                              app.ExampleName(), nuttx_chip),
+            runner=runner
+        )
+
+        self.argsOpt = []
+        self.chip_name = nuttx_chip
+        self.enable_frame_ptr = enable_frame_ptr
+        self.app = app
+        self.board = board
+
+    def generate(self):
+        self._Execute(['mkdir', '-p', self.output_dir],
+                      title='Generating ' + self.identifier)
+
+    def _build(self):
+        logging.info('Compiling NuttX %s at %s, ',
+                     NuttXTarget(self.board, self.app), self.output_dir)
+        nuttx_dir = os.path.join(os.sep, 'opt', 'nuttx', 'nuttx')
+
+        os.environ['PATH'] = os.path.join(os.sep, 'opt', 'nuttx', 'gcc-13', 'bin') + ':' + os.environ.get('PATH')
```

Comment:
You should not use python to change things directly. The reason is we use `--dry-run` to pring out the equivalent commands for building and this is why we use `_Execute` and similar for everything. Any python change as this above will not work when the commands are printed. This has to be somehow embedded inside execute.

You seem to try to give higher precedence to the nuttx compiler over other compilers. Other systems seem to do this by either activate scripts (e.g. esp32 uses a $IDF_PATH/export.sh) or by setting some internal paths that could override compiler settings (e.g. MBED_OS_PATH).

Reply:
Done.